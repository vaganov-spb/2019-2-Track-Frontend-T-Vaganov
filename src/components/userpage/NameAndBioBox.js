import React from 'react';
import { connect } from 'react-redux';
import PersonalFields from './PersonalFields';
import infoStyles from '../../styles/ProfileInfo.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

function NameAndBioBox(props) {
	if (!props.isLoaded) {
		return null;
	}

	return (
		<div className={infoStyles.user_fields}>
			<PersonalFields 
				name="Name" 
				placeholder="Enter your name!" 
				row='1'
				maxlen='25'
			/>
			<PersonalFields 
				name="Nick" 
				placeholder="Enter your username!" 
				description="Minimum lenght is 5 characters"
				row='1'
				maxlen='25'
			/>
			<PersonalFields 
				name="Bio" 
				description="Any details about you"
				row='3'
				maxlen='75'
			/>
		</div>
	);
}

const mapStateToProps = (state) => ({
	user: state.userinfo.info,
	error: state.userinfo.error,
	isLoaded: state.userinfo.isLoaded,
});

export default connect(mapStateToProps)( NameAndBioBox);