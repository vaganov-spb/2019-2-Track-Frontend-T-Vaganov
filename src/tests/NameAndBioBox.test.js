import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import NameAndBioBox from '../components/userpage/NameAndBioBox';

const mockStore = configureStore();

describe('NameAndBioBox renders and act correctly', () => {
	let store;
	let component;

	beforeEach(() => {
		store = mockStore({
			userinfo: {
				info: {
					Name: 'Borat',
					Nick: 'Sagdiev',
					Bio: 'Pasha Technik'
				},
				isLoaded: true,
				error: null, 
			}
		});
        
		component = renderer.create(
			<Provider store={store}>
				<MemoryRouter>
					<NameAndBioBox/>
				</MemoryRouter>  
			</Provider>
		);
	});

	it ('should render normally', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
});