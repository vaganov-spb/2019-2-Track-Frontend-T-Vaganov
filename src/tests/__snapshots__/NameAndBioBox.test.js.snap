// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameAndBioBox renders and act correctly should render normally 1`] = `
<div
  className="user_fields"
>
  <div
    className="wrapper"
  >
    <div
      className="field_name"
    >
       
      Name
       
    </div>
    <textarea
      className="name"
      maxLength="25"
      minLength="5"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Enter your name!"
      rows="1"
      value="Borat"
    />
    <div
      className="userlen"
    >
       
       
    </div>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="field_name"
    >
       
      Nick
       
    </div>
    <textarea
      className="name"
      maxLength="25"
      minLength="5"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Enter your username!"
      rows="1"
      value="Sagdiev"
    />
    <div
      className="userlen"
    >
       
      Minimum lenght is 5 characters
       
    </div>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="field_name"
    >
       
      Bio
       
    </div>
    <textarea
      className="name"
      maxLength="75"
      minLength="5"
      onChange={[Function]}
      onKeyPress={[Function]}
      rows="3"
      value="Pasha Technik"
    />
    <div
      className="userlen"
    >
       
      Any details about you
       
    </div>
  </div>
</div>
`;
